src/main/scala/it/pagopa/interop/purposeprocess/api/impl/PurposeApiServiceImpl.scala

```diff
@@ -734,6 +735,15 @@ final case class PurposeApiServiceImpl(
     else Future.failed(PurposeNotInDraftState(purpose.id))
   }
 
+  private def assertDailyCallsIsDifferentThanBefore(purpose: PersistentPurpose, dailyCalls: Int): Future[Unit] = {
+    val ordering: Ordering[OffsetDateTime] = Ordering(Ordering.by[OffsetDateTime, Long](_.toEpochSecond).reverse)
+    val previousDailyCalls                 = purpose.versions.sortBy(_.createdAt)(ordering).map(_.dailyCalls).headOption
+    previousDailyCalls match {
+      case Some(x) if x != dailyCalls => Future.successful(())
+      case _                          => Future.failed(DailyCallsEqualThanBefore(purpose.id))
```

Comment:
albeit similar, with this change we cover exactly the case we need (while leaving the absence of versions a responsibility of other checks)
```suggestion
      case Some(x) if x == dailyCalls => Future.failed(DailyCallsEqualThanBefore(purpose.id))
      case _                          => Future.successful(())
```